frontend/packages/db-structure/src/parser/schemarb/parser.ts

```diff
@@ -248,19 +252,27 @@ function extractDefaultValue(
   return null
 }
 
-function extractRelationshipTableNames(argNodes: Node[]): [string, string] {
+function extractRelationshipTableNames(
+  argNodes: Node[],
+): Result<[string, string], UnexpectedTokenWarningError> {
   const stringNodes = argNodes.filter((node) => node instanceof StringNode)
-  if (!(stringNodes.length === 2)) {
-    throw new Error('Foreign key relationship must have two table names')
+  if (stringNodes.length !== 2) {
+    return err(
+      new UnexpectedTokenWarningError(
+        'Foreign key relationship must have two table names',
+      ),
+    )
   }
 
-  const [foreignTableName, primaryTableName] = stringNodes.map((node) => {
-    // @ts-expect-error: unescaped is defined as string but it is actually object
-    if (node instanceof StringNode) return node.unescaped.value
-    return null
-  })
+  const [foreignTableName, primaryTableName] = stringNodes.map(
+    (node): string => {
+      // @ts-expect-error: unescaped is defined as string but it is actually object
+      if (node instanceof StringNode) return node.unescaped.value
+      return ''
+    },
+  ) as [string, string]
```

Comment:
An as cast was required here, because TypeScript cannot infer that it is a tuple of strings.